terraform/services/api-rds/main.tf

```diff
@@ -126,13 +169,13 @@ resource "aws_db_instance" "api" {
   db_subnet_group_name    = aws_db_subnet_group.subnet_group.name
   parameter_group_name    = aws_db_parameter_group.v16_parameter_group.name
   backup_retention_period = 7
-  iops                    = local.db_name == "ab2d-east-prod" ? "20000" : "5000"
+  iops                    = var.app == "bcda" ? "1000" : local.db_name == "ab2d-east-prod" ? "20000" : "5000"
   apply_immediately       = true
-  kms_key_id              = data.aws_kms_alias.main_kms.target_key_arn
-  multi_az                = local.db_name == "ab2d-east-prod"
-  vpc_security_group_ids  = [aws_security_group.sg_database.id]
-  username                = data.aws_secretsmanager_secret_version.database_user.secret_string
-  password                = data.aws_secretsmanager_secret_version.database_password.secret_string
+  kms_key_id              = var.app == "ab2d" && length(data.aws_kms_alias.main_kms) > 0 ? data.aws_kms_alias.main_kms[0].target_key_arn : null
+  multi_az                = var.app == "bcda" ? true : local.db_name == "ab2d-east-prod"
+  vpc_security_group_ids  = var.app == "bcda" ? concat([aws_security_group.sg_database.id], local.gedit_security_group_ids) : [aws_security_group.sg_database.id]
+  username                = var.app == "bcda" && length(data.aws_secretsmanager_secret_version.database_secret_version) > 0 ? jsondecode(data.aws_secretsmanager_secret_version.database_secret_version[0].secret_string)["username"] : var.app == "ab2d" && length(data.aws_secretsmanager_secret_version.database_user) > 0 ? data.aws_secretsmanager_secret_version.database_user[0].secret_string : null
+  password                = var.app == "bcda" && length(data.aws_secretsmanager_secret_version.database_secret_version) > 0 ? jsondecode(data.aws_secretsmanager_secret_version.database_secret_version[0].secret_string)["password"] : var.app == "ab2d" && length(data.aws_secretsmanager_secret_version.database_user) > 0 ? data.aws_secretsmanager_secret_version.database_user[0].secret_string : null
```

Comment:
The conditional on username and password is difficult to read and will only get worse if we continue in this direction with dpc. Instead, try creating username and password secrets for BCDA via the AWS console under this naming scheme: `${var.app}/${var.env}/db/username` and `${var.app}/${var.env}/db/password`. Use a ternary on the name for `data "aws_secretsmanager_secret" "db_password"` and `data "aws_secretsmanager_secret" "db_password"` resources for all three apps.